### DISCUSSION TITLE
How to set different layout for files and grep

### DISCUSSION THREAD

**Discussion by WilliamHsieh** [ibhagwan/fzf-lua#1332] [Q&A] [answered]
I would like to set a flex layout for file-related actions (files, git_files, oldfiles, etc.) and a horizontal layout for grep actions (live_grep, grep_word, etc.). Currently, there is only an option to set a default layout for everything, and the only way I've managed to achieve this is by manually adding winopts in each keymap through the api. Is there an alternative way to achieve this?

Thank you in advance!

**Comment by ibhagwan** [comment, marked as answer]
You can either do this in the keymap (as you have), or add `winopts` for each picker during setup, I.e. `files.winopts`, `grep.winopts` (which will affect all grep pickers), etc.

**Reply by WilliamHsieh** [reply]
Awesome! I didn't realize that winopts is available in each picker according to the documentation. 
Thank you so much!

**Reply by ibhagwan** [reply]
Most options are available from everywhere (setup, setup.defaults, picker, direct call):
https://github.com/ibhagwan/fzf-lua/blob/main/OPTIONS.md#general-usage
